xero/auth.py

```diff
@@ -391,3 +393,277 @@ def refresh(self):
+        >>>     return HttpResponseRedirect(authorization_url)
+        ...
+
+        `credentials.auth_state` should be stored by your app (e.g. in the database
+        or cache) as it will be used to complete the authorization during the callback.
+        The simplest way is to dump the whole credentials object by storing `credentials.state`
+
+     3) After authorization the user will be redirected to the callback URI
+        provided (e.g. https://example.com/oauth/xero/callback) along with a querystring
+        containing the authentication secret. Pass the full URI including querystring to
+        `verify()`, using the credentials constructed earlier:
+        
+        >>> def complete_xero_auth_view(request):
+        >>>     # Create the credentials object from wherever you stored the state
+        >>>     credentials = OAuth2Credentials(**cred_state)
+        >>>     # Then pass in the full URI the user was directed back to, including
+        >>>     # querystring
+        >>>     auth_secret = request.get_raw_uri()
+        >>>     credentials.verify(auth_secret)
+
+        A token will be fetched from Xero and saved as `credentials.token`. If the credentials
+        object needs to be created again either dump the whole object using:
+
+        >>> cred_state = credentials.state
+        >>> ...
+        >>> new_creds = OAuth2Credentials(**cred_state)
+
+        or just use the client_id, client_secret and token:
+
+        >>> token = credentials.token
+        >>> ...
+        >>> new_creds = OAuth2Credentials(client_id, client_secret, token=token)
+
+     4) Now the credentials may be used to authorize a Xero session. As OAuth2 allows
+        authentication for mulitple Xero Organisations, it is necessary to set
+        the tenant_id against which the xero client's queries will run.
+
+        >>> from xero import Xero
+        >>> # Use the first xero organisation (tenant) permitted
+        >>> credentials.set_default_tenant()
+        >>> xero = Xero(credentials)
+        >>> xero.contacts.all()
+        >>> ...
+        
+        To pick from multiple possible Xero organisations `tenant_id` may be set
+        explicitly:
+        
+        >>> tenants = credentials.get_tenants()
+        >>> credentials.tenant_id = tenants[1]['tenantId']
+        >>> xero = Xero(credentials)
+
+        `OAuth2Credentials.__init__()` accepts `tenant_id` as a keyword argument.
+
+     5) If a refresh token is available, it can be used to generate a new token:
+        >>> if credentials.expired():
+        >>>     credentials.refresh()
+        >>>     # Then store the new credentials or token somewhere for future use:
+        >>>     cred_state = credentials.state
+        >>>     # or
+        >>>     new_token = credentials.token
+
+        Note that in order for tokens to be refreshable, Xero API requires
+        `offline_access` to be included in the scope.
+    """
```

Comment:
The duplication of API documentation here is a little out of place. Having a docstring is a good idea; but it should refer the user to the relevant documentation section, rather than give a long form description here.